streams/client_test.go

```diff
@@ -210,143 +210,133 @@ func TestPublishEvents(t *testing.T) {
+			codecData:       [][]byte{[]byte("boom")},
+			codecErr:        []error{nil},
+			eventsToPublish: []publisher.Event{event},
+			remainingEvents: 0,
+		},
+		{ // Records with nil error codes should be ignored with some log
+			testname: "nil error codes",
+			putRecordsOut: []*kinesis.PutRecordsOutput{
+				{
+					Records: []*kinesis.PutRecordsResultEntry{
+						{ErrorCode: nil},
 					},
+					FailedRecordCount: aws.Int64(1),
 				},
-				FailedRecordCount: aws.Int64(1),
 			},
-		}
-		putRecordsErr := []error{nil}
-		client.streams = &StubClient{out: putRecordsOut, err: putRecordsErr}
-
-		rest, err := client.publishEvents(events)
-		if err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
-		if len(rest) != 0 {
-			t.Errorf("unexpected number of remaining events: %d", len(rest))
-		}
-	}
-
-	{
-		// Kinesis received the event but it was not persisted, probably due to underlying infrastructure failure
-		codecData := [][]byte{[]byte("boom")}
-		codecErr := []error{nil}
-		client.encoder = &StubCodec{dat: codecData, err: codecErr}
-
-		putRecordsOut := []*kinesis.PutRecordsOutput{
-			&kinesis.PutRecordsOutput{
-				Records: []*kinesis.PutRecordsResultEntry{
-					&kinesis.PutRecordsResultEntry{
-						ErrorCode: aws.String("simulated_error"),
+			putRecordsErr:   []error{nil},
+			codecData:       [][]byte{[]byte("boom")},
+			codecErr:        []error{nil},
+			eventsToPublish: []publisher.Event{event},
+			remainingEvents: 0,
+		},
+		{ // Kinesis received the event but it was not persisted, probably due to underlying infrastructure failure
+			testname: "event not persisted",
+			putRecordsOut: []*kinesis.PutRecordsOutput{
+				{
+					Records: []*kinesis.PutRecordsResultEntry{
+						{
+							ErrorCode: aws.String("simulated_error"),
+						},
 					},
+					FailedRecordCount: aws.Int64(1),
 				},
-				FailedRecordCount: aws.Int64(1),
 			},
-		}
-		putRecordsErr := []error{nil}
-		client.streams = &StubClient{out: putRecordsOut, err: putRecordsErr}
-
-		rest, err := client.publishEvents(events)
-		if err != nil {
-			t.Errorf("unexpected error: %v", err)
-		}
-		if len(rest) != 1 {
-			t.Errorf("unexpected number of remaining events: %d", len(rest))
-		}
+			putRecordsErr:   []error{nil},
+			codecData:       [][]byte{[]byte("boom")},
+			codecErr:        []error{nil},
+			eventsToPublish: []publisher.Event{event},
+			remainingEvents: 1,
+		},
+		{ // Kinesis received the event but it was not persisted, due to throughput exceeded error
+			testname: "throughput exceeded",
+			putRecordsOut: []*kinesis.PutRecordsOutput{
+				{
+					Records: []*kinesis.PutRecordsResultEntry{
+						{
+							ErrorCode: aws.String("ProvisionedThroughputExceededException"),
+						},
+					},
+					FailedRecordCount: aws.Int64(1),
+				},
+			},
+			putRecordsErr:   []error{nil},
+			codecData:       [][]byte{[]byte("boom")},
+			codecErr:        []error{nil},
+			eventsToPublish: []publisher.Event{event},
+			remainingEvents: 1,
```

Comment:
Can you add add prefixes so it will be easier to distinguish between the data and the expected behavior?
For example: `expectedRemainingEvents`, `mockPutRecordsErr`, `mockCodecData`.

Reply:
done